Userland/Libraries/LibArchive/Zip.cpp

```diff
@@ -138,6 +140,63 @@ ErrorOr<void> ZipOutputStream::add_member(ZipMember const& member)
     return local_file_header.write(*m_stream);
 }
 
+ErrorOr<double> ZipOutputStream::add_member_from_stream(StringView path, Stream& stream, Optional<Core::DateTime> optional_date)
```

Comment:
Returning a `double` to represent the compression ratio seems like a pretty straightforward copy/paste from `zip`. But maybe a nicer API is to return a struct with some information about the added entry, including the compression ratio and compressed size?

Also, maybe a `double` is a bit overkill to represent a ratio that is usually displayed with low precision - a `float` will probably do as well.